app/src/main/java/com/example/zingmp3phake/presenter/DetailSongActivityPresenter.kt

```diff
@@ -0,0 +1,159 @@
+                    handler.removeCallbacksAndMessages(null)
+                    handler.postDelayed({
+                        getCurrentSong()
+                        getTimeforView()
+                    }, TIME_DELAY_FOR_LOAD)
+                }
+                else -> {}
+            }
+        }
+    }
+
+    private fun handlePlayOrPause() {
+        detailSongView.displayPlayOrPause(musicService.isPlayings)
+    }
+
+    override fun bindService(context: Context) {
+        this.context = context
+        if (isConnected == false) {
+            val service = Intent(context, MusicService::class.java)
+            context.bindService(service, serviceConnection, Context.BIND_AUTO_CREATE)
+        }
+    }
+
+    override fun unBindService() {
+        if (isConnected) {
+            context?.unbindService(serviceConnection)
+        }
+    }
+
+    override fun getCurrentSong() {
+        if (isConnected == false) {
+            context?.let { bindService(it) }
+        } else {
+            val song = musicService.listSongs.get(musicService.positions)
+            detailSongView.displayCurrentSong(song)
+        }
+    }
+
+    override fun handleChangeSeekBar(value: Int) {
+        musicService.onChangeSeekBar(value)
+    }
+
+    override fun getLyrics() {
+        val song = musicService.listSongs.get(musicService.positions)
+        if (song.lyrics.equals(N0_LYRIC)) detailSongView.displayLyricSong(mutableListOf())
+        else {
+            songRepo.remote.getLyricSong(
+                song.songInfo.songid,
+                object : Listener<MutableList<String>> {
+                    override fun onSuccess(listLine: MutableList<String>) {
+                        detailSongView.displayLyricSong(listLine)
+                    }
+
+                    override fun onFail(msg: String) {
+                        detailSongView.displayLyricSong(mutableListOf())
+                    }
+                }
+            )
+        }
+    }
+
+    override fun handleEventSong(event: String) {
+        when (event) {
+            MusicAction.FAVORITE.name -> {
+                val song = musicService.listSongs.get(musicService.positions)
+                detailSongView.displayFavorite(song.isFavorite)
+                getPendingIntent(MusicAction.FAVORITE.name)?.send()
+            }
+            else -> getPendingIntent(event)?.send()
+        }
+    }
+
+    private fun getTimeforView() {
+        handler.post(object : Runnable {
+            override fun run() {
+                val time = musicService?.getCurrentSongTime()
+                if (time != null) {
+                    detailSongView.displayCurrentTimeSong(time)
+                } else detailSongView.displayCurrentTimeSong(0)
+                handler.removeCallbacks(this)
+                handler.postDelayed(this, TIME_SLEEP_100)
+            }
+        })
+    }
+
+    private fun getPendingIntent(action: String): PendingIntent? {
```

Comment:
```suggestion
    private fun getPendingIntent(action: String): PendingIntent? {
```
logic này thuộc về view mình phải handle bên view e
Mà chỗ này em trả về ``PendingIntent`` mà không thấy dùng e nhỉ?